### PR TITLE
[Merged by Bors] - refactor(Topology/Category): add CompHausLike.Limits

### PR DESCRIPTION
**Pull request by dagurtomas**
This is the first part of the refactor of CompHaus and friends (adding the new API which will eventually be used to redefine the categories of compact Hausdorff spaces, see https://github.com/leanprover-community/mathlib4/pull/12930)

---
- [x] depends on: #13904 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Category/CompHausLike/Limits.lean [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+/-- The functor to `TopCat` preserves pullbacks. -/
+noncomputable instance : PreservesLimit (cospan f g) (compHausLikeToTop P) :=
+  preservesLimitOfCreatesLimitAndHasLimit _ _
+
+/-- The functor to another `CompHausLike` preserves pullbacks. -/
+noncomputable instance {P' : TopCat → Prop}
+    (h : ∀ (X : CompHausLike P), P X.toTop → P' X.toTop) :
+    PreservesLimit (cospan f g) (toCompHausLike h) := by
+  have : PreservesLimit (cospan f g) (toCompHausLike h ⋙ compHausLikeToTop P') :=
+    show PreservesLimit _ (compHausLikeToTop _) from inferInstance
```
Comment (dagurtomas) [inline Mathlib/Topology/Category/CompHausLike/Limits.lean:280]:
```suggestion
    inferInstanceAs (PreservesLimit _ (compHausLikeToTop _))
```

File: Mathlib/Topology/Category/CompHausLike/Limits.lean [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+    preservesColimitOfIsoDiagram _ Discrete.natIsoFunctor.symm
+  apply preservesColimitOfPreservesColimitCocone (CompHausLike.finiteCoproduct.isColimit _)
+  exact TopCat.sigmaCofanIsColimit _
+
+/-- The functor to another `CompHausLike` preserves finite coproducts if they exist. -/
+noncomputable instance {P' : TopCat.{u} → Prop}
+    (h : ∀ (X : CompHausLike P), P X.toTop → P' X.toTop) :
+    PreservesFiniteCoproducts (toCompHausLike h) := by
+  have : PreservesFiniteCoproducts (toCompHausLike h ⋙ compHausLikeToTop P') :=
+    show PreservesFiniteCoproducts (compHausLikeToTop _) from inferInstance
```
Comment (dagurtomas) [inline Mathlib/Topology/Category/CompHausLike/Limits.lean:167]:
```suggestion
    inferInstanceAs (PreservesFiniteCoproducts (compHausLikeToTop _))
```

**Comment by dagurtomas** [comment]
(Sorry if everyone got github notifications for some unrelated commit, I was merging master and messed up...)

File: Mathlib/Topology/Category/CompHausLike/Limits.lean [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+      (_ : OpenEmbedding f), HasExplicitPullback f g) :
+    HasExplicitPullbacksOfInclusions P :=
+  { hasProp := by
+      intro _ _ _ f
+      apply hP'
+      exact Sigma.openEmbedding_ι _ _ }
+
+/-- The functor to `TopCat` creates pullbacks of inclusions if they exist. -/
+noncomputable instance [HasExplicitPullbacksOfInclusions P] :
+    PreservesPullbacksOfInclusions (compHausLikeToTop P) :=
```
Comment (jcommelin) [inline Mathlib/Topology/Category/CompHausLike/Limits.lean:329]: The Lean statement doesn't seem to be about creating pullbacks, only preserving them.

**Comment by jcommelin** [comment]
bors d+

**Comment by dagurtomas** [comment]
Thanks!

bors merge

**Pull request closed without merging**